spccore/internal/lock.py

```diff
@@ -27,18 +30,22 @@
 
 LOCK_DEFAULT_MAX_AGE = datetime.timedelta(seconds=10)
 DEFAULT_BLOCKING_TIMEOUT = datetime.timedelta(seconds=70)
-CACHE_UNLOCK_WAIT_TIME = 0.5
+CACHE_UNLOCK_WAIT_TIME_SEC = 0.5
+LOCK_FILE_SUFFIX = 'lock'
 
 
-class LockedException(Exception):
+class LockException(Exception):
     pass
 
 
 class Lock(object):
     """
     Implements a lock by making a directory named <lock_name>.lock
+
+    Notes: we found that the OS operation for checking the file exists and creating a file is not guarantee the
+    transactional boundary across platform. However, checking for directory name exists and creating it has
+    transactional boundary across platform. Therefore, we use directory instead of file.
```

Comment:
Notes: There is no cross-platform, atomic OS operation that both checks whether a file exists and creates the file. However, the directory creation OS operation is atomic across platforms.  Therefore we use a directory rather than a file as the semaphore to lock file access.